### DISCUSSION TITLE
Invalid position in decode

### DISCUSSION THREAD

**Discussion by UnderPalms** [john30/ebusd#910] [Q&A]
I have several Vaillant-devices connected to the Bus. For all I can read the values except for the solar station (circuit "sc").
When I read "SolGain" I get "invalid position in decode". When I promt "ebusctl find Gain" the output is (amongst others):
"sc SolGain (ERR: invalid position for 31ecb509030d0300 / 00"

Output for "ebusctl info" is: 

version: ebusd 23.1.23.1-9-g1895339e
update check: revision 23.1 available, device firmware 1[8419] available
device: /dev/ttyAMA0, enhanced, invalid, firmware 1.1[3501].1[3501]
access: *
signal: acquired
symbol rate: 23
max symbol rate: 172
min arbitration micros: 17
max arbitration micros: 22
min symbol latency: 4
max symbol latency: 4
reconnects: 0
masters: 6
messages: 918
conditional: 18
poll: 1
update: 10
address 00: master #1
address 01: master #6
address 03: master #11
address 06: slave #6, scanned "MF=Vaillant;ID=PMS02;SW=0209;HW=8402", loaded "va                                                                             illant/06.pms.csv"
address 08: slave #11, scanned "MF=Vaillant;ID=BAI00;SW=0104;HW=7803", loaded "v                                                                             aillant/bai.308523.inc", "vaillant/08.bai.csv"
address 0a: slave, scanned "MF=Vaillant;ID=PMW01;SW=0206;HW=8302", loaded "vaill                                                                             ant/0a.pmw.hwc.csv"
address 10: master #2
address 12: slave, scanned "MF=Vaillant;ID=PMW01;SW=0206;HW=8302"
address 15: slave #2, scanned "MF=Vaillant;ID=70000;SW=0613;HW=6903", loaded "va                                                                             illant/15.700.csv"
address 26: slave, scanned "MF=Vaillant;ID=VR_71;SW=0104;HW=0503", loaded "vaill                                                                             ant/26.vr_71.csv"
address 31: master #8, ebusd
address 36: slave #8, ebusd, scanning
address ec: slave, scanned "MF=Vaillant;ID=SOL00;SW=0613;HW=6903", loaded "vaill                                                                             ant/ec.sol.sc.csv"
address f7: master #20
address fc: slave #20, scanned "MF=Vaillant;ID=PMW01;SW=0206;HW=8302"

has anyone an idea about the error "invalid position in decode" and how it can be resolved?
Thanks upfront for any comment,
Peter

**Comment by avanmourik** [comment]
It may be totally unrelated, but I have been checking the influence of latency on messages on the output of ebusctl find.
for 50 ms: 
hmu YieldThisYear10 =  (ERR: invalid position for 3108b516041a010a00 / 00)
hmu YieldThisYear11 =  (ERR: invalid position for 3108b516041a010b00 / 00)
hmu YieldThisYear12 =  (ERR: invalid position for 3108b516041a010c00 / 00)
and for 150 ms:
hmu YieldThisYear10 = no data stored			
hmu YieldThisYear11 = no data stored			
hmu YieldThisYear12 = no data stored.

In fact all ERR messages of this kind disappeared after adjusting latency.
The no data stored relates to another bigger issue

**Reply by UnderPalms** [reply]
Thanks a lot. For which latency value it finally worden?

**Reply by avanmourik** [reply]
I have now changed the config file to use 150 ms

**Reply by UnderPalms** [reply]
I tried to vary the latency but no change in the situation.

**Comment by john30** [comment]
the device answers with a zero length, so it just does not support the message. no data stored just means that you didn't get an answer at all

**Reply by UnderPalms** [reply]
If it is the case that the device does not Support the message, is there any way to try automatically several different commands? Like a kind of "bruteforce"
